### PR TITLE
The Mouthbay Update - Dentisty

### PR DESCRIPTION
**Pull request by CodenameB**
This PR will make the following changes:

Add two new surgeries to the Medbay arsenal. Tooth replacement, and tooth extraction.
The bioprinter can now print generic teeth to insert into a mob.
Gold can now be used to craft gold teeth, these have a new sprite. No other special properties.
An alternate title for doctors has been added, "Dentist"
There is a minor bugfix with dragging objects that affected teeth and caused unrelated stacks to merge.

Suggested followup: Allow replacing teeth with gold to change a mob's desc_modifier to reflect this. This would require some significant changes to the teeth code that I don't think I'm up to.

Thanks to @Kurfursten on github for the idea. Thanks to Shadowmech88 on discord for coding help and the bugfix.

:cl:
 * rscadd: Two new surgeries added. One to remove teeth (bonesetter -> hemostat -> bonesetter) and one to replace teeth (scalpel -> FixoVein -> teeth -> cautery)
* rscadd: The bioprinter can now print teeth.
 * rscadd: You can now make gold teeth out of gold. These act the same as any other teeth.

### CODE REVIEW

**Comment by Shadowmech88** [comment]
When you insert a gold tooth into someone's mouth, does it turn into a regular tooth, or can it be extracted as a gold tooth?

**Comment by CodenameB** [comment]
@Shadowmech88 

Right now, it very simply increases the patient's teeth counter by one. Adding a gold teeth counter (or different species teeth counter) would involve greatly changing the way teeth are currently coded. Right now, as far as I can tell, they are just a number var. The species and such aren't applied to the teeth until after they pop out, the obj doesn't exist before that. They aren't like organs at all.

I might be misreading the code entirely however.

**Comment by Shadowmech88** [comment]
I wonder if it's worth having 32 individual tooth datums for each human mob, just to keep track of all the properties of their teeth.

In a future PR, of course.

**Comment by Kurfursten** [comment]
Since you don't take tooth damage to an individual tooth (Rather, any one tooth has a chance of popping out), it seems like just a counter for gold teeth would be fine (then you could calculate a chance of a gold tooth falling out).

Neat though. 👍 Perhaps another logical extension of this is that a random ailment someone can get (like appendicitus) is "tooth ache" which causes pain but goes away if a tooth is lost (by any means).

**Comment by Blithering** [comment]
inserting xeno/goliath/catbeast teeth to increase bite damage when?

**Comment by PJB3005** [comment]
I don't think the dentist alt title is deserved. 👍 to the rest though.

**Comment by DamianX** [comment]
You're going to need a changelog

**Comment by CodenameB** [comment]
@PJB3005
I understand your concern regarding the alt title, here is my reasoning for adding it:

The thought behind the alt title is that the dental job will grow to meet the title. I had the same thoughts as @Kurfursten regarding a random illness (I was thinking an abscessed tooth) but coding it seemed intimidating since it might involve those aforementioned 32 tooth datums. Once we add implanting teeth from other races (to give yourself fangs or extra strong bites, etc) and add some stuff like a debuff to "sweet" food that causes cavities if too much is eaten (Thus ensuring chefs can't spam ice cream and other cheap food, and to make vending machines less appealing) then we would need full time dentists.

Adding the title now puts in a reminder to work on that, and adds motivation.

**Comment by PJB3005** [comment]
No, not really.

First I find your point about teeth cavities dumb. Such a feature would cause only infuriation.

If we go that route, add dentist later, not now.

All the other alt titles somewhat make sense. Dentist? Not really. They can stick with surgeon.

File: code/game/machinery/bioprinter.dm [hunk 1]
```diff
@@ -69,7 +70,8 @@
 		stored_matter -= products[choice][2]
 		var/new_organ = products[choice][1]
 		var/obj/item/organ/O = new new_organ(get_turf(src))
-		O.is_printed = TRUE // Heist stuff.
+		if(choice == !"tooth")
```
Comment (PJB3005) [inline code/game/machinery/bioprinter.dm]: 1. This doesn't work, you want `!=`
2. Terrible hack. Do something else.

File: code/modules/surgery/tooth.dm [hunk 2]
```diff
@@ -0,0 +1,221 @@
+//Procedures in this file: Tooth replacement surgery, Tooth extraction surgery
+//////////////////////////////////////////////////////////////////
+//						TOOTH REPLACEMENT SURGERY							//
+//////////////////////////////////////////////////////////////////
+
+/datum/surgery_step/tooth
+	priority = 10
+	can_infect = 0
+	can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
```
Comment (PJB3005) [inline code/modules/surgery/tooth.dm]: No relative pathing please.

File: code/modules/surgery/tooth.dm [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+		if (!hasorgans(target))
+			return 0
+		var/datum/organ/external/affected = target.get_organ(target_zone)
+		if (!affected)
+			return 0
+		return target_zone == "mouth"
```
Comment (PJB3005) [inline code/modules/surgery/tooth.dm]: Use defines here.

File: code/modules/surgery/tooth.dm [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+
+//////////////////////////////////////////////////////////////////
+//						TOOTH EXTRACTION SURGERY							//
+//////////////////////////////////////////////////////////////////
+
+/datum/surgery_step/extract
+	priority = 5
+	can_infect = 0
+	can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
```
Comment (PJB3005) [inline code/modules/surgery/tooth.dm]: No relative pathing.

File: code/modules/surgery/tooth.dm [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+	user.visible_message("<span class='warning'>[user]'s hand slips, cutting [target]'s gums with \a [tool]!</span>", \
+	"<span class='warning'>Your hand slips, cutting [target]'s gums with \a [tool]!</span>")
+	target.apply_damage(5, BRUTE, affected)
+
+
+//////////////////////////////////////////////////////////////////
+//						TOOTH EXTRACTION SURGERY							//
+//////////////////////////////////////////////////////////////////
+
+/datum/surgery_step/extract
```
Comment (PJB3005) [inline code/modules/surgery/tooth.dm]: Change the name, it's too generic.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/modules/surgery/tooth.dm (hunk 2 above)
Comment (DamianX) [reply code/modules/surgery/tooth.dm]: It's like this on all surgeries, just saying.

Comment (PJB3005) [reply code/modules/surgery/tooth.dm]: So?

Comment (DamianX) [reply code/modules/surgery/tooth.dm]: So we may want to fix all of them

File: code/modules/surgery/tooth.dm (hunk 4 above)
Comment (CodenameB) [reply code/modules/surgery/tooth.dm]: @PJB3005 
This is the way all of the surgeries are coded. I literally lifted that line directly out of the face surgery code. If you want to go and fix every single surgery, by all means go ahead. I'm not going to.

File: code/game/machinery/bioprinter.dm (hunk 1 above)
Comment (CodenameB) [reply code/game/machinery/bioprinter.dm]: This line is merely to make it not produce runtimes. That part of the code works fine. I can change it to != though. What do you mean by "something else" ? I am not much of a coder, so I need specifics please.

File: code/modules/surgery/tooth.dm (hunk 4 above)
Comment (Shadowmech88) [reply code/modules/surgery/tooth.dm]: I believe he wants you to just fix this surgery.

File: code/game/machinery/bioprinter.dm (hunk 1 above)
Comment (PJB3005) [reply code/game/machinery/bioprinter.dm]: The code is assuming every thing printed is an organ, which is why you need this hack. Rewrite it so it checks whether the printed thing is an organ before assuming it's one.

File: code/modules/surgery/tooth.dm (hunk 3 above)
Comment (CodenameB) [reply code/modules/surgery/tooth.dm]: Gonna need some help on the best way, if any, to do this.

**Comment by CodenameB** [comment]
@PJB3005 I believe I have made the changes you requested.

**Pull request merged**